### PR TITLE
Always test against default distribution when in a FIPS JVM

### LINKED ISSUE #50364: Distribution tests failing due to unknown setting [xpack.security.ssl.diagnose.trust]
```process was found dead while waiting for ports files, node{:distribution:archives:integ-test-zip:integTest-0}```

Caused by:
```

ERROR][o.e.b.ElasticsearchUncaughtExceptionHandler] [integTest-0] uncaught exception in thread [main] |  
-- | --
  | »  org.elasticsearch.bootstrap.StartupException: java.lang.IllegalArgumentException: unknown setting [xpack.security.ssl.diagnose.trust] please check that any required plugins are installed, or check the breaking changes documentation for removed settings |  
  | »  	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:174) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT] |  
  | »  	at org.elasticsearch.bootstrap.Elasticsearch.execute(Elasticsearch.java:161) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT] |  
  | »  	at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:86) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT] |  
  | »  	at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:125) ~[elasticsearch-cli-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT] |  
  | »  	at org.elasticsearch.cli.Command.main(Command.java:90) ~[elasticsearch-cli-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT] |  
  | »  	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:126) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT] |  
  | »  	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:92) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT] |  
  | »  Caused by: java.lang.IllegalArgumentException: unknown setting [xpack.security.ssl.diagnose.trust] please check that any required plugins are installed, or check the breaking changes documentation for removed settings |  
  | »  	at org.elasticsearch.common.settings.AbstractScopedSettings.validate(AbstractScopedSettings.java:530) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT] |  
  | »  	at org.elasticsearch.common.settings.AbstractScopedSettings.validate(AbstractScopedSettings.java:475) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT] |  
  | »  	at org.elasticsearch.common.settings.AbstractScopedSettings.validate(AbstractScopedSettings.java:446) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT] |  
  | »  	at org.elasticsearch.common.settings.AbstractScopedSettings.validate(AbstractScopedSettings.java:417) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT] |  
  | »  	at org.elasticsearch.common.settings.SettingsModule.<init>(SettingsModule.java:149) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT] |  
  | »  	at org.elasticsearch.node.Node.<init>(Node.java:352) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT] |  
  | »  	at org.elasticsearch.node.Node.<init>(Node.java:253) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT] |  
  | »  	at org.elasticsearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:221) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT] |  
  | »  	at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:221) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT] |  
  | »  	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:349) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT] |  
  | »  	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:170) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT] |  
  | »  	... 6 more
```

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+matrix-java-periodic/ES_BUILD_JAVA=openjdk13,ES_RUNTIME_JAVA=java8fips,nodes=general-purpose/397/console
https://gradle-enterprise.elastic.co/s/rghajdqd44lle

Related to #49900?

### PR DESCRIPTION
**Pull request by mark-vieira**
Running with a FIPS JVM is only supported when using the default distribution, therefore, it doesn't make sense for us to test against the integ-test or OSS distributions when FIPS mode is enabled.

Closes https://github.com/elastic/elasticsearch/issues/50364

### CODE REVIEW

**Comment by mark-vieira** [comment]
@elasticmachine update branch

File: plugins/examples/painless-whitelist/build.gradle [hunk 1]
```diff
@@ -33,7 +35,7 @@ dependencies {
 }
 
 testClusters.integTest {
-  testDistribution = 'oss'
+  testDistribution = BuildParams.inFipsJvm ? 'DEFAULT' : 'oss'
```
Comment (rjernst) [inline plugins/examples/painless-whitelist/build.gradle]: Why the different casing?

**Review by rjernst — APPROVED** [review]
LGTM

File: plugins/examples/painless-whitelist/build.gradle (hunk 1 above)
Comment (mark-vieira) [reply plugins/examples/painless-whitelist/build.gradle]: Groovy is lenient here. I've made this consistently uppercase.

**Review by jkakavas — APPROVED** [review]
LGTM

**Pull request merged**